File: src/com/l2kt/gameserver/network/clientpackets/RequestGetBossRecord.java
package com.l2kt.gameserver.network.clientpackets;

import java.util.Map;

import com.l2kt.gameserver.data.manager.RaidPointManager;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.serverpackets.ExGetBossRecord;

public class RequestGetBossRecord extends L2GameClientPacket
{
	@SuppressWarnings("unused")
	private int _bossId;
	
	@Override
	protected void readImpl()
	{
		_bossId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player player = getClient().getActiveChar();
		if (player == null)
			return;
		
		final int points = RaidPointManager.getInstance().getPointsByOwnerId(player.getObjectId());
		final int ranking = RaidPointManager.getInstance().calculateRanking(player.getObjectId());
		final Map<Integer, Integer> list = RaidPointManager.getInstance().getList(player);
		
		player.sendPacket(new ExGetBossRecord(ranking, points, list));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestGetBossRecord.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.manager.RaidPointManager
import com.l2kt.gameserver.network.serverpackets.ExGetBossRecord

class RequestGetBossRecord : L2GameClientPacket() {
    private var _bossId: Int = 0

    override fun readImpl() {
        _bossId = readD()
    }

    override fun runImpl() {
        val player = client.activeChar ?: return

        val points = RaidPointManager.getPointsByOwnerId(player.objectId)
        val ranking = RaidPointManager.calculateRanking(player.objectId)
        val list = RaidPointManager.getList(player)

        player.sendPacket(ExGetBossRecord(ranking, points, list))
    }
}